### ISSUE TITLE
Type error when create a dict with both int and float values

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#327]
**Describe the bug**
Please see this example.
![image](https://user-images.githubusercontent.com/3479453/68269386-9e58c900-0094-11ea-9e1a-cc04e4dcb018.png)

When initialize a dict with both `int` and `float` parameter, it reports a "no overloads" error.
Same thing happens if not use literial values:

![image](https://user-images.githubusercontent.com/3479453/68269519-07404100-0095-11ea-8e43-34a6eaa0b559.png)

Currently this kind of problem was only found when using int and float together, not sure if there's other type pair can cause this.

Thanks!

**Code**
```python
a = dict(x=1.1, y=1.1)
b = dict(x=1, y=1)
c = dict(x=1, y=1.1)
```
```python
m = int(1)
n = float(1.1)
a = dict(x=m, y=m)
b = dict(x=n, y=n)
c = dict(x=m, y=n)
```

**VS Code extension or command-line**
VS Code extension 1.0.83

**Comment by erictraut** [comment]
Thanks for the bug report. This was a long-standing issue in the TypeVar matching logic. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is now fixed in 1.0.84, which I just published.
